unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/WearableGridController.cs

```diff
@@ -0,0 +1,162 @@
+
+            view.SetWearableBreadcrumb(wearableBreadcrumbModel with { ResultCount = resultCount });
+        }
+
+        private void HandleNewPageRequested(int page)
+        {
+            requestWearablesCancellationToken = requestWearablesCancellationToken.SafeRestart();
+            RequestWearablesAndShowThem(page, requestWearablesCancellationToken.Token).Forget();
+        }
+
+        private async UniTask<int> RequestWearablesAndShowThem(int page, CancellationToken cancellationToken)
+        {
+            UserProfile ownUserProfile = userProfileBridge.GetOwn();
+            string ownUserId = ownUserProfile.userId;
+
+            try
+            {
+                // TODO: instead of requesting owned wearables, we should request all the wearables with the current filters & sorting
+                (IReadOnlyList<WearableItem> wearables, int totalAmount) = await wearablesCatalogService.RequestOwnedWearablesAsync(
+                    ownUserId,
+
+                    // page is not zero based
+                    page + 1,
+                    PAGE_SIZE, true, cancellationToken);
+
+                var wearableModels = wearables.Select(wearable =>
+                {
+                    if (!Enum.TryParse(wearable.rarity, true, out NftRarity rarity))
+                    {
+                        rarity = NftRarity.Common;
+                        Debug.LogError($"Could not parse the rarity of the wearable: {wearable.rarity}. Fallback to common..");
+                    }
+
+                    return new WearableGridItemModel
+                    {
+                        WearableId = wearable.id,
+                        Rarity = rarity,
+                        ImageUrl = wearable.ComposeThumbnailUrl(),
+                        IsEquipped = ownUserProfile.HasEquipped(wearable.id),
+
+                        // TODO: make the new state work
+                        IsNew = false,
+                        IsSelected = false,
+                    };
+                });
+
+                view.SetWearablePages(page, (totalAmount / PAGE_SIZE) + 1);
+
+                // TODO: mark the wearables to be disposed if no references left
+                view.ClearWearables();
+                view.ShowWearables(wearableModels);
+
+                return totalAmount;
+            }
+            catch (Exception e) { Debug.LogException(e); }
+
+            return 0;
+        }
+
+        private void HandleWearableSelected(WearableGridItemModel wearableGridItem)
+        {
+            view.ClearWearableSelection();
+            view.SelectWearable(wearableGridItem.WearableId);
+        }
+
+        private void HandleWearableUnequipped(WearableGridItemModel wearableGridItem)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void HandleWearableEquipped(WearableGridItemModel wearableGridItem)
+        {
+            throw new NotImplementedException();
```

Comment:
I guess we're going to leave this part for a separated task, right?

Reply:
For the moment yes, so we dont delay this pr anymore